designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ScenarioActionRepository.scala

```diff
@@ -68,7 +120,12 @@ class DbScenarioActionRepository(
 
   import profile.api._
 
-  def addInProgressAction(
+  override def executeCriticalSection[T](dbioAction: DB[T]): DB[T] = for {
```

Comment:
WDYT about moving it out of the class to some specialized trait? I would see the trait like this:

```scala
trait LocableTable {
   protected def table: LTableQuery[_]

  def withLockedTable[T](
      dbioAction: DB[T]
  ): DB[T] = {...}

}

WDYT?

Reply:
done